pkg/logger/dataframe.go

```diff
@@ -28,38 +28,36 @@ const (
 	StderrStreamTag
 )
 
-const headerLength = 8
+const HeaderLength = 8
 
 type DataFrame struct {
 	Tag  StreamTag
 	Size int
 	Data []byte
 }
 
-var EmptyDataFrame DataFrame
-
-func NewDataFrameFromReader(reader io.Reader) (DataFrame, error) {
-	header := make([]byte, headerLength)
+func NewDataFrameFromReader(reader io.Reader) (*DataFrame, error) {
+	header := make([]byte, HeaderLength)
 
 	n, err := reader.Read(header)
 	if err != nil {
-		return DataFrame{}, err
+		return nil, err
 	}
-	if n != headerLength {
-		return DataFrame{}, fmt.Errorf("unable to read dataframe header")
+	if n != HeaderLength {
+		return nil, fmt.Errorf("unable to read dataframe header")
 	}
 
-	df := DataFrame{}
+	df := &DataFrame{}
 	df.Tag = StreamTag(binary.LittleEndian.Uint32(header))
 	df.Size = int(binary.BigEndian.Uint32(header[4:]))
 	df.Data = make([]byte, df.Size)
 
 	n, err = reader.Read(df.Data)
 	if err != nil {
-		return DataFrame{}, err
+		return nil, err
```

Comment:
This seems to be returning `EOF` before the container is done, related to my comment about `job logs` not working for me locally.